upup/pkg/kutil/rollingupdate_cluster.go

```diff
@@ -288,10 +287,11 @@ func (n *CloudInstanceGroup) RollingUpdate(cloud fi.Cloud, force bool, interval
 
 		// TODO: Batch termination, like a rolling-update
 
-		request := &ec2.TerminateInstancesInput{
-			InstanceIds: []*string{u.ASGInstance.InstanceId},
+		request := &autoscaling.TerminateInstanceInAutoScalingGroupInput{
+			InstanceId:                     u.ASGInstance.InstanceId,
+			ShouldDecrementDesiredCapacity: aws.Bool(false),
 		}
-		_, err := c.EC2().TerminateInstances(request)
+		_, err := c.Autoscaling().TerminateInstanceInAutoScalingGroup(request)
```

Comment:
Personally I think we should be doing this. But the outcome is the same. Main reason for switching to the autoscaling api is for better testability since the mockec2 and mockautoscaling implementations aren't integrated together yet.